snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/event/AbstractEvent.kt

```diff
@@ -58,12 +61,24 @@ abstract class AbstractEvent : Event {
      * @return the event custom context entities
      */
     override val entities: MutableList<SelfDescribingJson>
-        get() = customContexts
+        get() {
+            if (isProcessing) {
+                if (entitiesForProcessing != null) {
```

Comment:
Nit: just realized now that since we are calling `entitiesForProcessing` here twice (this line and the next), it results in the entities list being generated twice. Maybe we could avoid that (e.g., store in a local variable or use `?.let`)?